src/transformers/models/msclap/configuration_msclap.py

```diff
@@ -0,0 +1,489 @@
+            Whether or not the model should return the last key/values attentions (not used by all models). Only
+            relevant if `config.is_decoder=True`.
+        bos_token_id (`int`, *optional*, defaults to 50256):
+            The beginning of sequence token ID.
+        eos_token_id (`int`, *optional*, defaults to 50256):
+            The end of sequence token ID.
+        scale_attn_by_inverse_layer_idx (`bool`, *optional*, defaults to `False`):
+            Whether to scale the attention weights by the inverse of their layer index.
+        reorder_and_upcast_attn (`bool`, *optional*, defaults to `False`):
+            Whether to reorder and upcast the attention.
+        projection_dim (`int`, *optional*, defaults to 768):
+            Dimension of the projection head of the `ClapTextModelWithProjection`.
+
+    Examples:
+
+    ```python
+    >>> from transformers import MSClapTextConfig, MSClapTextModel
+
+    >>> # Initializing a MSCLAP text configuration
+    >>> configuration = MSClapTextConfig()
+
+    >>> # Initializing a model (with random weights) from the configuration
+    >>> model = MSClapTextModel(configuration)
+
+    >>> # Accessing the model configuration
+    >>> configuration = model.config
+    ```"""
+
+    model_type = "msclap_text_model"
+
+    def __init__(
+        self,
+        vocab_size=50257,
+        hidden_size=768,
+        projection_dropout_prob=0,
+        initializer_factor=1.0,
+        n_positions=1024,
+        n_embd=768,
+        num_hidden_layers=12,
+        num_attention_heads=12,
+        n_inner=None,
+        activation_function="gelu_new",
+        resid_pdrop=0.1,
+        embd_pdrop=0.1,
+        attn_pdrop=0.1,
+        layer_norm_epsilon=1e-5,
+        initializer_range=0.02,
+        summary_type="cls_index",
+        summary_use_proj=True,
+        summary_activation=None,
+        summary_proj_to_labels=True,
+        summary_first_dropout=0.1,
+        scale_attn_weights=True,
+        use_cache=True,
+        bos_token_id=50256,
+        eos_token_id=50256,
+        scale_attn_by_inverse_layer_idx=False,
+        reorder_and_upcast_attn=False,
+        projection_dim=768,
+        **kwargs,
+    ):
+        super().__init__(**kwargs)
+
+        self.text_model_config = GPT2Config(
+            vocab_size,
+            n_positions,
+            n_embd,
+            num_hidden_layers,
+            num_attention_heads,
+            n_inner,
+            activation_function,
+            resid_pdrop,
+            embd_pdrop,
+            attn_pdrop,
+            layer_norm_epsilon,
+            initializer_range,
+            summary_type,
+            summary_use_proj,
+            summary_activation,
+            summary_proj_to_labels,
+            summary_first_dropout,
+            scale_attn_weights,
+            use_cache,
+            bos_token_id,
+            eos_token_id,
+            scale_attn_by_inverse_layer_idx,
+            reorder_and_upcast_attn,
+        )
```

Comment:
Could we use `AutoConfig` here? I'm also wondering whether we really need to pass all of the GPT2 config arguments here!

You could instead pass a dictionnary that has every GPT2 arguments, the same way you've done below for the text config. WDYT ?